src/core/server/saved_objects/migrations/actions/integration_tests/actions.test.ts

```diff
@@ -598,6 +598,48 @@ describe('migration actions', () => {
         }
       `);
     });
+    it('resolves left with unavailable_shards_exception if a shard is no longer available', async () => {
```

Comment:
I tried a few different scenarios, and struggled to actually reproduce this exception. Currently the test is simply attempting to clone into an index after a node shutdown has been initiated.

I also tried running a transform since the related cloud SDH mentioned that nodes having persistent tasks assigned to them (like transform jobs) could experience these issues when they were already shutting down.

In both cases I couldn't get to an `unavailable_shards_exception`, rather, this test would fail and no errors would be caught (this was after I patched an ES client bug I found, see my other comment below):

<img width="948" alt="Screen Shot 2022-06-22 at 1 37 35 PM" src="https://user-images.githubusercontent.com/1608770/175133927-58f284b9-a6c1-4eb1-ac30-b5c991c6eed9.png">

It's very possible I'm overlooking something here as I'm not intimately familiar with the origins of this particular issue. But it also seems the reported problem occurred as a result of a handful of different behaviors, so maybe a variety of conditions are necessary to reproduce it?

@Rudolf [mentioned](https://github.com/elastic/kibana/issues/127136#issuecomment-1135218992) that we still saw occurrences of this even after https://github.com/elastic/elasticsearch/pull/85277 was addressed, so in theory we should still be able to reproduce it...